controller_manager:
  ros__parameters:
    update_rate: 500 # Hz

    left_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    
    right_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController

left_arm_controller:
  ros__parameters:
    command_interfaces:
      - position
    state_interfaces:
      - position
    joints:
      - left_shoulder_pitch_joint
      - left_shoulder_roll_joint
      - left_shoulder_yaw_joint
      - left_elbow_joint
      - left_wrist_roll_joint
      - left_wrist_pitch_joint
      - left_wrist_yaw_joint
    
    state_publish_rate: 1.0
    action_monitor_rate: 500.0
    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01

right_arm_controller:
  ros__parameters:
    command_interfaces:
      - position
    state_interfaces:
      - position
    joints:
      - right_shoulder_pitch_joint
      - right_shoulder_roll_joint
      - right_shoulder_yaw_joint
      - right_elbow_joint
      - right_wrist_roll_joint
      - right_wrist_pitch_joint
      - right_wrist_yaw_joint

    state_publish_rate: 1.0
    action_monitor_rate: 500.0
    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true
    constraints:
      stopped_velocity_tolerance: 0.01
